aten/src/ATen/autocast_mode.cpp

```diff
@@ -360,6 +360,10 @@ Therefore, for the moment, this is all copy pasted in from VariableTypeEverythin
   m.impl(REGISTER_NAME, \
     &WrapFunction<CastPolicy::POLICY, SIGNATURE, SIGNATURE, &FUNC>::type::call);
 
+#define KERNEL_LEGACY_SIGNATURE(FUNC, REGISTER_NAME, SIGNATURE, LEGACY_SIGNATURE, POLICY) \
+  m.impl(REGISTER_NAME, \
+    c10::impl::hacky_wrapper_for_legacy_signatures<SIGNATURE>(TORCH_FN((&WrapFunction<CastPolicy::POLICY, LEGACY_SIGNATURE, LEGACY_SIGNATURE, &FUNC>::type::call))));
```

Comment:
I think this can be directly be eliminated. There aren't N functions you have to port: all you have to do is modify `cached_cast` to handle `optional<Tensor>`. Actually, I'm not sure if AMP properly supported undefined tensors previously anyway (cc @mcarilli )

Reply:
this is eliminated in the next PR stacked on top.